CBLClient/PredictiveQueries.py

```diff
@@ -0,0 +1,68 @@
+from CBLClient.Args import Args
+from CBLClient.Client import Client
+
+
+class PredictiveQueries(object):
+    _db = None
+    _baseUrl = None
+
+    def __init__(self, base_url):
+        self.base_url = base_url
+
+        # If no base url was specified, raise an exception
+        if not self.base_url:
+            raise Exception("No base_url specified")
+
+        self._client = Client(base_url)
+
+    def registerModel(self, modelName):
+        args = Args()
+        args.setString("model_name", modelName)
+        return self._client.invokeMethod("predictiveQuery_registerModel", args)
+
+    def unregisterModel(self, modelName):
+        args = Args()
+        args.setString("model_name", modelName)
+        return self._client.invokeMethod("predictiveQuery_unRegisterModel", args)
+
+    def getPredictionQueryResult(self, model, dictionary, database):
+        args = Args()
+        args.setMemoryPointer("model", model)
+        args.setMemoryPointer("dictionary", dictionary)
+        args.setMemoryPointer("database", database)
+        return self._client.invokeMethod("predictiveQuery_getPredictionQueryResult", args)
+
+    def queryNonDictionaryInput(self, model, dictionary, database):
+        args = Args()
+        args.setMemoryPointer("model", model)
+        args.setString("dictionary", dictionary)
+        args.setMemoryPointer("database", database)
+        return self._client.invokeMethod("predictiveQuery_nonDictionary", args)
+
+    def getNumberOfCalls(self, model):
+        args = Args()
+        args.setMemoryPointer("model", model)
+        return self._client.invokeMethod("predictiveQuery_getNumberOfCalls", args)
+
+    def euclideanDistance(self, database, key1, key2):
+        args = Args()
+        args.setMemoryPointer("database", database)
+        args.setString("key1", key1)
+        args.setString("key2", key2)
+        return self._client.invokeMethod("predictiveQuery_euclideanDistance", args)
+
+    def squaredEuclideanDistance(self, database, key1, key2):
```

Comment:
would be better to prefix it with get or set

Reply:
done